pkg/harvester/components/novnc/NovncConsole.vue

```diff
@@ -35,28 +44,52 @@ export default {
 
   data() {
     return {
-      rfb:          null,
-      connected:    false,
-      disconnected: false,
+      rfb:               null,
+      connected:         false,
+      disconnected:      false,
+      reconnectDelay:    3000,
+      reconnecting:      false,
+      reconnectingTimes: 0,
     };
   },
 
   mounted() {
     this.$nextTick(() => {
+      this.connect();
+    });
+  },
+
+  methods: {
+    connect() {
       const rfb = new RFB(this.$refs.view, this.url);
 
       rfb.addEventListener('connect', () => {
         this.connected = true;
+        this.reconnectingTimes = 0;
+        this.reconnecting = false;
       });
+
       rfb.addEventListener('disconnect', (e) => {
         this.disconnected = true;
+        this.rfb = null;
+        this.reconnect();
       });
 
       this.rfb = rfb;
-    });
-  },
+    },
+
+    reconnect() {
+      this.reconnectingTimes += 1;
+      this.reconnecting = true;
+      this.connected = false;
+      this.disconnected = false;
+      console.log('reconnecting vnc'); // eslint-disable-line no-console
+
+      setTimeout(() => {
+        this.connect();
+      }, this.reconnectDelay);
+    },
 
-  methods: {
     disconnect() {
```

Comment:
Not related to current change - I think this can be removed, unless it's not called by any parent component.